src/agent/debuglet.js

```diff
@@ -263,30 +264,31 @@ Debuglet.createDebuggee =
   return new Debuggee(properties);
 };
 
-    /**
-     * @private
-     */
-    Debuglet.prototype.getProjectId_ = function(callback) {
+/**
+ * @private
+ */
+Debuglet.prototype.getProjectId_ = function(callback) {
   var that = this;
 
   // We need to figure out whether we are running on GCP. We can use our ability
   // to access the metadata service as a test for that.
   // TODO: change this to getProjectId in the future.
-  common.utils.getProjectNumber(function(err, metadataProject) {
-    // We should get an error if we are not on GCP.
-    var onGCP = !err;
-
-    // We perfer to use the locally available projectId as that is least
-    // surprising to users.
-    var project = that.debug_.options.projectId || process.env.GCLOUD_PROJECT ||
-                  metadataProject;
-
-    // We if don't have a projectId by now, we fail with an error.
-    if (!project) {
-      return callback(err);
-    }
-    return callback(null, project, onGCP);
-  });
+  metadata.project(
+      'numeric-project-id', function(err, response, metadataProject) {
```

Comment:
My thought was to expose it as a constant within our own module but if it's only ever referenced here that's fine